File: seed/rust-sdk/trace/src/types/submission_status_for_test_case.rs
use crate::test_case_result_with_stdout::TestCaseResultWithStdout;
use crate::test_case_grade::TestCaseGrade;
use crate::traced_test_case::TracedTestCase;
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
#[serde(tag = "type")]
pub enum SubmissionStatusForTestCase {
        Graded {
            #[serde(flatten)]
            data: TestCaseResultWithStdout,
        },

        GradedV2 {
            value: TestCaseGrade,
        },

        Traced {
            #[serde(flatten)]
            data: TracedTestCase,
        },
}

File: seed/ruby-sdk-v2/trace/lib/seed/submission/submission_status_for_test_case.rb
# frozen_string_literal: true

module Seed
    module Types
        class SubmissionStatusForTestCase < Internal::Types::Union

            discriminant :type

            member -> { Seed::Submission::TestCaseResultWithStdout }, key: "GRADED"
            member -> { Seed::Submission::TestCaseGrade }, key: "GRADED_V_2"
            member -> { Seed::Submission::TracedTestCase }, key: "TRACED"
    end
end
